main/samba/add_missing___compar_fn_t.patch

```diff
@@ -0,0 +1,14 @@
+--- a/source4/dsdb/samdb/ldb_modules/count_attrs.c
++++ b/source4/dsdb/samdb/ldb_modules/count_attrs.c
+@@ -38,6 +38,11 @@
+ #define NULL_REQ_PSEUDO_N -2LL;
+ #define STAR_REQ_PSEUDO_N -4LL;
+ 
++#ifndef __COMPAR_FN_T
++#define __COMPAR_FN_T
++typedef int (*__compar_fn_t)(const void *, const void *);
++#endif
++
```

Comment:
the `__comar_fn_t` is a glibc internal. The proper way to use qsort is to use `int (*)(const void *, const void *)`.

```diff
diff --git a/source4/dsdb/samdb/ldb_modules/count_attrs.c b/source4/dsdb/samdb/ldb_modules/count_attrs.c
index b0d1b22..29f1b94 100644
--- a/source4/dsdb/samdb/ldb_modules/count_attrs.c
+++ b/source4/dsdb/samdb/ldb_modules/count_attrs.c
@@ -222,7 +222,7 @@ static const char **get_sorted_attrs(TALLOC_CTX *mem_ctx,
 		attrs[i] = a;
 	}
 
-	qsort(attrs, n_attrs, sizeof(char *), (__compar_fn_t)strcasecmp_ptr);
+	qsort(attrs, n_attrs, sizeof(char *), (int (*)(const void *, const void *))strcasecmp_ptr);
 	return attrs;
 }
 
@@ -313,7 +313,7 @@ static int count_attrs_search_callback(struct ldb_request *req,
 			}
 
 			qsort(found_attrs, msg->num_elements, sizeof(char *),
-			      (__compar_fn_t)strcasecmp_ptr);
+			      (int (*)(const void *, const void *))strcasecmp_ptr);
 
 
 			/* find and report duplicates */
```

See https://pubs.opengroup.org/onlinepubs/9699919799/functions/qsort.html

This should be fixed upstream though.